README.md

```diff
@@ -13,175 +13,413 @@ Compartmentalization of host behavior hooks for JS.
+    record: StaticModuleRecord,
+
+    // Properties to copy to the `import.meta` of the resulting module instance,
+    // if the source mentions `import.meta`.
+    importMeta?: Object,
+  }
+
+  // Describes a module by a *reusable* third-party static module record.
+  // When the compartment _loads_ the module, it will use the bindings array
+  // of the third-party static module record to discover shallow dependencies
+  // in lieu of compiling the source, and otherwise behaves identically
+  // to { source } descriptors.
+  // When the compartment _imports_ the module, it will construct a
+  // [[ModuleEnvironmentRecord]] and [[ModuleExportsNamespace]] based
+  // entirely on the bindings array of the third-party static module record.
+  // If the third-party static-module record has a true `needsImportMeta`
+  // property, the compartment will construct an `importMeta` by the
+  // same process as any { source } module.
+  // `importMeta` will otherwise be `undefined`.
+  // The compartment will then initialize the module by calling the
+  // `initialize` function of the third-party static module record,
+  // giving it the module environment record and `importMeta`.
+  // The compartment memoizes a promise for the module exports namespace
+  // that is fulfilled as already specified in 262 for dynamic import,
+  // where the behavior if the initialize function is async is analogous to
+  // top-level-await for a module compiled from source.
+  | {
+    record: ThirdPartyStaticModuleRecord,
+
+    // Properties to copy to the `import.meta` of the resulting module instance,
+    // if the `record` has a `true` `needsImportMeta` property.
+    importMeta?: Object,
+  }
+
+  // Use a static module record from the compartment in which this compartment
+  // was constructed.
+  // If the compartment _loads_ the corresponding module, the static
+  // module record will be synchronously available only if it is already
+  // in the parent compartment's synchronous memo.
+  // Otherwise, loading induces the parent compartment to load the module,
+  // but does not induce the parent compartment to load the module's transitive
+  // dependencies.
+  // This compartment then resolves the shallow dependencies according
+  // to its own resolveHook and loads the consequent transitive dependencies.
+  // If the compartment _imports_ the module, the behavior is equivalent
+  // to loading for the retrieved static module record or third-party static
+  // module record.
+  | {
+    record: string,
+
+    // Properties to copy to the `import.meta` of the resulting module instance.
+    importMeta?: Object,
+  }
+
+  // FOR SHARED MODULE INSTANCES:
+
+  // To create an alias to an existing module instance in this compartment:
+  | {
+    // A full specifier for another module in this compartment.
+    instance: string,
+  }
+
+  // To create an alias to an existing module instance given the full specifier
+  // of the module in a different compartment:
```

Comment:
What if the target compartment does not have the sync memo of this spec? Does it call the hook to load the module?